frontend/src/test/java/starter/project/frontend/CarControllerTest.java

```diff
@@ -1,21 +1,56 @@
 package starter.project.frontend;
 
-import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.web.client.TestRestTemplate;
-import org.springframework.boot.web.server.LocalServerPort;
-import org.springframework.http.ResponseEntity;
+import starter.project.domain.Car;
+import starter.project.domain.Doors;
+import starter.project.domain.Engine;
+import starter.project.domain.Wheels;
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
 
-import java.net.URL;
-import static org.junit.jupiter.api.Assertions.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
 
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class CarControllerTest {
-    @LocalServerPort
-    private int port;}
-//    @Autowired
-//    //private TestRestTemplate restTemplate
+
+    CarController controller = new CarController();
+
+    @Test
+    public void newCarsAAreAddedToMap() {
+
+        Car car = Car.builder().id(15).build();
+        controller.addCar(car);
+        assertEquals(1, controller.map.size());
+        Car secondCar = Car.builder().id(13).build();
+        controller.addCar(secondCar);
+        System.out.println(controller.map);
+        assertEquals(2, controller.map.size());
```

Comment:
We can have some kind of formatting where we do all of our logic and then all of our assertions. But this looks good! We could also check that cars are added correctly to the map by attempting to retrieve them.